modules/ocf_desktop/files/suspend/ocf-suspend

```diff
@@ -1,28 +1,83 @@
-#!/bin/bash
-# Suspend the machine and wake on LAN.
-
-# Don't sleep if...
-# installing packages
-! /usr/bin/lsof -- /var/lib/dpkg/lock > /dev/null 2>&1 || exit 1
-# puppet is running
-[ ! -f /var/lib/puppet/state/agent_catalog_run.lock ] || exit 2
-# user present
-[ $(w --no-header | wc -l) -eq 0 ] || exit 3
-# is a virtual machine
-imvirt | grep -qi Physical || exit 4
-
-# Try to unmount /home and /tmp (both tmpfs)
-if grep $'tmpfs\t/home' /etc/fstab > /dev/null; then
-	umount /home 2> /dev/null && mount /home || exit 255
-fi
-umount /tmp 2> /dev/null && mount /tmp
-
-# Check if a kernel update is needed, restart if so
-if [ -f /var/run/apt-dater-host_reboot ]; then
-    rm -f /var/run/apt-dater-host_reboot
-    shutdown -r now > /dev/null 2>&1
-fi
-
-# Enable WOL and sleep until magic packet wakes you up
-/sbin/ethtool -s eth0 wol g
-/usr/sbin/pm-suspend > /dev/null
+#!/usr/bin/env python3
+"""Suspend this desktop if not in use and lab is closed.
+
+Contains a few checks (is anyone logged in, is Puppet running, etc.) before
+shutting off.
+
+If a kernel update is needed, the desktop will restart instead of suspending.
+"""
+import argparse
+import os
+import subprocess
+import sys
+from subprocess import DEVNULL
+
+from ocflib.lab.hours import Day
+
+
+def file_is_open(path):
+    """Return whether a file is open by any process on the system."""
+    return subprocess.call(('lsof', '--', path), stdout=DEVNULL, stderr=DEVNULL) == 0
+
+
+def main(argv=None):
+    parser = argparse.ArgumentParser(description=__doc__)
+    parser.add_argument('--force', '-f', action='store_true', help='force suspend, even if the lab is open')
+    parser.add_argument('--quiet', '-q', action='store_true', help='stay quiet except for unexpected errors')
+    args = parser.parse_args(argv)
+
+    def report(line):
+        if not args.quiet:
+            print('[ocf-suspend] {}'.format(line), file=sys.stderr)
```

Comment:
...wait, do we have other commands doing it this way? Most programs I've seen use the `programname: errormessage` format